Pod/Classes/SEGAnalytics.m

```diff
@@ -158,8 +158,19 @@ - (void)handleAppStateNotification:(NSNotification *)note
                             };
     });
     SEL selector = NSSelectorFromString(selectorMapping[note.name]);
-    if (selector)
-        [self callIntegrationsWithSelector:selector arguments:nil options:nil sync:true];
+  if (selector) {
+    if ([note.name isEqualToString:UIApplicationDidFinishLaunchingNotification]) {
+      while (!self.initialized) {
+        [[NSRunLoop currentRunLoop] runMode:NSDefaultRunLoopMode beforeDate:[NSDate dateWithTimeIntervalSinceNow: 1.0]];
+      }
+    }
+    NSArray *arguments = nil;
+    if (note != nil) {
+      arguments = @[note];
```

Comment:
I think those definitions are fine. I updated my code so a notification is passed in instead of a dictionary.